lib/hls/hls_parser.js

```diff
@@ -985,14 +990,56 @@ shaka.hls.HlsParser = class {
       goog.asserts.assert(streamInfos.audio.length ||
           streamInfos.video.length, 'We should have created a stream!');
 
+      const drmInfos = [];
+      const keyIds = new Set();
+
+      if (sessionKeyTags.length > 0) {
+        const characteristics = tag.getAttributeValue('CHARACTERISTICS');
+        for (const drmTag of sessionKeyTags) {
+          const drmCharacteristics =
+              drmTag.getAttributeValue('CHARACTERISTICS');
+          if ((characteristics && !drmCharacteristics) ||
```

Comment:
I don't understand what this long check is doing.  What does CHARACTERISTICS have to do with EXT-X-SESSION-KEY?

Reply:
Changed!